src/funcx_common/task_storage/s3.py

```diff
@@ -49,39 +57,41 @@ def _store_to_s3(self, task: TaskProtocol, result: str) -> None:
             )
         except botocore.exceptions.ClientError as err:
             raise StorageException(
-                f"Putting result into s3 for task:{task.task_id} failed"
+                f"Putting {kind.name} into s3 for task:{task.task_id} failed"
             ) from err
         else:
-            task.result_reference = {
+            reference = {
                 "storage_id": "s3",
                 "s3bucket": self.bucket_name,
                 "key": key,
             }
+            setattr(task, f"{kind.name}_reference", reference)
```

Comment:
There are an awful lot of references to `{kind.name}_reference`. Regardless of whether it becomes `kind.value` or not, I think this should be wrapped in a helper.

It can be attached to the enum class, e.g.
```python
class StorageFieldName(enum.Enum):
    result = "result"
    payload = "payload"

    @property
    def reference_attr(self):
        return f"{self.value}_reference"
```
Usage works like you'd expect:
```python
>>> StorageFieldName.payload
<StorageFieldName.payload: 'payload'>
>>> StorageFieldName.payload.reference_attr
'payload_reference'
```

Reply:
Makes sense.